salt/modules/zypper.py

```diff
@@ -788,16 +789,23 @@ def mod_repo(repo, **kwargs):
 
     if kwargs.get('gpgautoimport') is True:
         global_cmd_opt.append('--gpg-auto-import-keys')
+        trigger_refresh = True
 
     if cmd_opt:
         cmd_opt = global_cmd_opt + ['mr'] + cmd_opt + [repo]
         __zypper__.refreshable.xml.call(*cmd_opt)
 
     # If repo nor added neither modified, error should be thrown
-    if not added and not cmd_opt:
+    if not added and not cmd_opt and not trigger_refresh:
```

Comment:
@isbm 
```python
if kwargs.get('gpgautoimport'):
```
is wrong because we don't want this:
```yaml
repo_with_gpg:
  pkgrepo.managed:
    - url: http://some.url/path
    - gpgautoimport: any-string-here
```